core/include/scipp/core/parallel-tbb.h

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#ifndef SCIPP_CORE_PARALLEL_H
+#define SCIPP_CORE_PARALLEL_H
+
+#include <tbb/parallel_for.h>
+
+#include "scipp/common/index.h"
+
+/// Wrappers for multi-threading using TBB.
+namespace scipp::core::parallel {
+
+inline auto blocked_range(const scipp::index begin, const scipp::index end,
+                          const scipp::index grainsize = -1) {
+  // TBB's default grain-size is 1, which is probably quite inefficient in
+  // some cases, in particular given the slow random-access of ViewIndex. A
+  // good default value is not known right now. In practice this should also
```

Comment:
Does Mantid have a good guess for a default value?

Reply:
I don't think Mantid uses `tbb::parallel_for`. Honestly I have no clue why TBB uses this. I would have thought cutting the range into pieces as big as possible would be most sensible, but maybe it is not for other workloads.

The default I put here is probably not what will be used ultimately, the callers should set a better grainsize manually. This should be simple to figure out for dense data. For sparse data however we need to take into account that for certain techniques different regions of the detectors see vastly different event rates, i.e., we have to avoid using to large grains, since this would imply that even if each thread processes the same number of pixels they will have to process potentially very different numbers of events.

Reply:
> I don't think Mantid uses `tbb::parallel_for`

It does, but only in a couple of places, OpenMP is the go-to. And I'm fairly sure it never touches the default grin size.

Reply:
>     * It would be useful to put some developer instructions in place for the `tbb` setup (maybe just in the README for now)

@OwenArnold I updated the readme, see https://github.com/scipp/scipp/blob/master/README.md#tbb.

Reply:
>  I would have thought cutting the range into pieces as big as possible would be most sensible, but maybe it is not for other workloads.

Maybe it's trying to do something like OpenMP's scheduling where by default it doesn't divide the work in to large equal pieces but a mix of large and small pieces and allocates to threads on a first come first serve basis? I am just guessing here, I don't know much about TBB, never used it before.

Reply:
No, it actually makes just lots of size-1 pieces, I had had a look when implementing this.